include/mxnet/tensor_blob.h

```diff
@@ -442,7 +442,7 @@ class FieldEntry<mxnet::TShape>
         throw dmlc::ParamError(os.str());
     }
     if (enforce_nonzero_) {
-      for (mxnet::index_t i = 0; i < v.ndim(); ++i) {
+      for (int i = 0; i < v.ndim(); ++i) {
```

Comment:
why int here?

Reply:
dimensions will not be hight and `int` is used across many template for dim such as https://github.com/dmlc/mshadow/blob/95ebe0f109ae021d0d66e2a627ccfc55c3253b55/mshadow/tensor.h#L43